src/Language/PureScript/Ide/Types.hs

```diff
@@ -245,6 +248,16 @@ identifierFromDeclarationRef = \case
   P.TypeOpRef _ op -> P.showOp op
   _ -> ""
 
+declarationType :: IdeDeclaration -> DeclarationType
```

Comment:
I had to move this out to avoid a cyclic import. This might not be the best place for it but it was my best guess.

Reply:
It's good enough :D The whole `Types` module shouldn't exist, but it's so convenient compared to figuring out all the intra dependencies.